rust/phoenix-channel/src/lib.rs

```diff
@@ -131,6 +131,8 @@ pub enum Error {
     TokenExpired,
     #[error("max retries reached")]
     MaxRetriesReached,
+    #[error("login failed")]
```

Comment:
Would a more descriptive error be more fitting?

```suggestion
    #[error("join 'login' topic failed")]
```

Reply:
The topic name isn't login but client, gateway or relay! :)

Printing that isn't very useful though because it is always the component you are running, i.e. a client most of the time.